Plugins/Bugsnag/Source/Bugsnag/Private/Android/AndroidThread.h

```diff
@@ -22,17 +22,30 @@ class FAndroidThread : public IBugsnagThread, FJavaObjectWrapper
 
 	FString GetId() const override
 	{
-		auto Id = (*Env).CallLongMethod(JavaObject, Cache->ThreadGetId);
-		FAndroidPlatformJNI::CheckAndClearException(Env);
-		return Id ? FString::Printf(TEXT("{Id}"), Id) : TEXT("");
+		// Retrieve the thread ID as a Java string
+		jstring jId = static_cast<jstring>((*Env).CallObjectMethod(JavaObject, Cache->ThreadGetId));
+		if (FAndroidPlatformJNI::CheckAndClearException(Env) || !jId)
+		{
+			return TEXT("");
+		}
+
+		// Convert the Java string to an FString
+		const char* nativeString = (*Env).GetStringUTFChars(jId, nullptr);
+		FString Id = FString(UTF8_TO_TCHAR(nativeString));
```

Comment:
Should we maybe be null-checking `nativeString` as well? There are conditions where `GetStringUTFChars` will return null for a valid string.